spec/service_worker/index.html

```diff
@@ -3589,6 +3632,148 @@ <h2 id="status">Status of This Document</h2>
     </spec-algorithm>
   </spec-section>
 
+  <spec-section id="foreign-fetch-scope-match-algorithm">
+    <h1>Match Service Worker for Foreign Fetch</h1>
```

Comment:
I think your proposal is probably the saner resolution. This way foreign fetch interception is strictly subsidiary to same-origin fetch interception interception and it's always trivial to share handler logic for requests where the calling origin is not important.